### PR TITLE
feat: market place bucket

### LINKED ISSUE #355: Provide terraform for bucket creation
Bucket policy

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "DelegateS3Access",
            "Effect": "Allow",
            "Principal": {
                "AWS": ""
            },
            "Action": [
                "s3:ListBucket",
                "s3:GetObject*",
                "s3:PutObject*"
            ],
            "Resource": [
                "arn:aws:s3:::sps-dev-ds-storage/*",
                "arn:aws:s3:::sps-dev-ds-storage"
            ]
        }
    ]
}
```
AWS account value will be available in SSM to be created by UCS

Documentation provided by UCS on how to add to marketplace is available at https://unity-sds.gitbook.io/docs/developer-docs/common-services/docs/users-guide/deployment/deployment-concepts-and-infrastructure/unity-marketplace

### PR DESCRIPTION
**Pull request by wphyojpl**
Closes #355 

NOTE: It requires an update to [uds-sbx-cumulus_lambda_processing_policy](https://us-east-1.console.aws.amazon.com/iam/home?region=us-east-1#/roles/details/uds-sbx-cumulus-lambda-processing/editPolicy/uds-sbx-cumulus_lambda_processing_policy?step=addPermissions) as noted in the other ticket. 

Requires the following variables. 

          prefix  = "uds-test-cumulus"
          market_bucket_name = "william_test_1"
          market_bucket__notification_prefix = "stage_out"
          tags = {
            Venue = "test",
            ServiceArea = "ds",
            CapVersion = "18.0.0"
            Component = "Cumulus",
            Proj = "Unity",
            CreatedBy = "ds",
            Env = "test",
            Stack = "Cumulus"
          }
          uds_aws_account_ssm_path="/unity/uds/account"
          uds_aws_account_region_ssm_path="/unity/uds/account/region""
          uds_prefix_ssm_path="/unity/uds/prefix"

It requires 3 SSM parameters for UDS AWS account, account region, and prefix

It creates a bucket with the [name](https://us-west-2.console.aws.amazon.com/s3/buckets/uds-sbx-cumulus-unity-william-test-1?region=us-west-2&bucketType=general&tab=permissions) "uds-sbx-cumulus-unity-william-test-1".

### CODE REVIEW

**Comment by wphyojpl** [comment]
Cross Account [Test](https://us-west-2.console.aws.amazon.com/s3/buckets/unity-dev-cumulus-unity-william-test-1?region=us-west-2&bucketType=general&prefix=URN%3ANASA%3AUNITY%3AUDS_LOCAL_TEST%3ADEV%3AUDS_COLLECTION___2404251100/URN%3ANASA%3AUNITY%3AUDS_LOCAL_TEST%3ADEV%3AUDS_COLLECTION___2404251100%3Aabcd.1234.efgh.test_file05/&showversions=false)

**Comment by wphyojpl** [comment]
Hi @ngachung 

[PR](https://github.com/unity-sds/unity-marketplace/pull/6) is created. But I think [variable line](https://github.com/unity-sds/unity-marketplace/blob/8f901f1900271f44406e3f0177c7fb29977ebf70/applications/unity-ds/7.7.0/metadata.json#L20) is wrong.

File: tf-module/marketplace/variables.tf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  description = "name of S3 bucket. Note-1: it will be prefixed with '<project prefix>-unity-'. Note-2: It should only have '-'. '_' will be replaced with '-'"
+}
+variable "market_bucket__notification_prefix" {
+  type = string
+  default = "stage_out"
+  description = "path to the directory where catalogs.json will be written"
+}
+variable "uds_aws_account_ssm_path" {
+  type = string
+  default = "/unity/uds/account"
```
Comment (ngachung) [inline tf-module/marketplace/variables.tf:20]: Default to /unity/shared-services/aws/account

**Comment by wphyojpl** [comment]
Hi @ngachung 
Looks like marketplace PR is merged. 
I guess this is done, right?

**Comment by wphyojpl** [comment]
Need to update the structure based on [this link](https://unity-sds.gitbook.io/docs/developer-docs/common-services/docs/users-guide/deployment/deployment-concepts-and-infrastructure/unity-marketplace) by Anil
Not sure it is finalized at this moment.

**Review by ngachung — APPROVED** [review]

**Pull request merged**